pennylane/pulse/rydberg.py

```diff
@@ -0,0 +1,125 @@
+
+        Args:
+            qubit_positions (list): list of coordinates of the Rydberg atoms
+            separation (float): separation between the Rydberg atoms (in meters)
+            wires (list): List of wires containing the wire values for all the atoms. This list should
+                have the same length as ``qubit_positions``.
+
+        Returns:
+            Sum: hamiltonian of the interaction
+        """
+        if self._rydberg_interaction is not None:
+            return self._rydberg_interaction
+
+        def rydberg_projector(wire: int) -> SProd:
+            """Returns the projector into the Rydberg state for the given wire.
+
+            Args:
+                wire (int): _description_
+
+            Returns:
+                SProd: projector into the Rydberg state
+            """
+            return (1 - qml.PauliZ(wire)) / 2
+
+        coeffs = []
+        ops = []
+        for idx, (pos1, wire1) in enumerate(zip(self.qubit_positions[1:], self.wires[1:])):
+            for pos2, wire2 in zip(self.qubit_positions[: idx + 1], self.wires[: idx + 1]):
+                atom_distance = np.linalg.norm(qml.math.array(pos2) - pos1)
+                Vij = 1 / (abs(atom_distance) ** 6)  # van der Waals potential
+                coeffs.append(Vij)
+                ops.append(qml.prod(rydberg_projector(wire1), rydberg_projector(wire2)))
+
+        self._rydberg_interaction = qml.dot(coeffs, ops)
+        return self._rydberg_interaction
+
+    def drive(self, amplitudes: list, detunings: list, wires: list) -> ParametrizedHamiltonian:
+        r"""Returns a :class:`ParametrizedHamiltonian` describing the evolution of the Rydberg ensemble
+        when driving the given atoms (``wires``) with lasers with the corresponding ``amplitude``
+        and ``detuning``:
+
+        .. math::
+
+            H = \frac{\hbar}{2} \sum_i  \Omega_i(t) \sigma_i^x - \frac{\hbar}{2} \sum_i
+                \delta_i(t) \sigma_i^z + \sum_{i<j} U_{ij} n_i n_j
+
+        where :math:`\Omega_i` and :math:`\delta_i` correspond to the amplitude and detuning of the
+        laser applied to atom :math:`i`, and :math:`\sigma^\alpha` for :math:`\alpha = x,y,z` are
+        the Pauli matrices.
+
+        The last term of the sum corresponds to the interaction between the atoms of the ensemble
+        due to the Rydberg blockade. For more info, check :seealso:`rydberg_interaction`.
+
+        .. note::
+
+            :math:`\hbar` is set to 1.
+
+        Args:
+            amplitudes (list): list of laser amplitudes
+            detunings (list): list of laser detuning parameters
+            wires (list): list of wires to drive
+
+        Returns:
+            ParametrizedHamiltonian: hamiltonian describing the laser driving
+        """
+        H1 = 1 / 2 * qml.dot(amplitudes, [qml.PauliX(w) for w in wires])
+        H2 = -1 / 2 * qml.dot(detunings, [qml.PauliZ(w) for w in wires])
+        return H1 + H2 + self.rydberg_interaction
```

Comment:
I think we should keep `rydberg_interaction` out of a method called `drive`, as it is part of an always-on drift term

Reply:
If we remove the `rydberg_interaction` from `drive`, it will never get queued.

I added this here because, as you already mentioned, this is an intrinsic parameter that we should ALWAYS take into account. 

As explained in the docstring, `drive` returns the Hamiltonian of the Rydberg ensemble when driving it with lasers, thus I feel we should return the rydberg interaction. If needed, we could add a `driving_term` property that returns only the driving term.